libcanard/canard.c

```diff
@@ -849,10 +849,20 @@ CANARD_PRIVATE int8_t rxSessionUpdate(CanardInstance* const          ins,
     }
     else
     {
+        // The purpose of the correct_start check is to reduce the possibility of accepting a malformed multi-frame
+        // transfer in the event of a CRC collision. The scenario where this failure mode would manifest is as follows:
+        // 1. A valid transfer (whether single- or multi-frame) is accepted with TID=X.
+        // 2. All frames of the subsequent multi-frame transfer with TID=X+1 are lost except for the last one.
+        // 3. The CRC of said multi-frame transfer happens to yield the correct residue when applied to the fragment
+        //    of the payload contained in the last frame of the transfer (a CRC collision is in effect).
+        // 4. The last frame of the multi-frame transfer is erroneously accepted even though it is malformed.
+        // The correct_start check eliminates this failure mode by ensuring that the first frame is observed.
```

Comment:
Yes. One (or more) frame gets through, then some frames are lost.

Reply:
What's special about the first frame then? Why is ensuring you have the first frame different from ensuring you have at least two frames with different toggle bits?

Reply:
The first transfer is needed to bring the transfer reassembly state machine into the state where it is initialized and ready to accept the next transfer. Without the first successful transfer, the transfer reassembly state machine will reside in the initialization state, where it will keep rejecting all incoming frames until it sees the one where the start-of-transfer flag is set.